tests/unit/test_database.py

```diff
@@ -319,9 +319,8 @@ def test_database_save(tdb, start_scmrun):
     assert len(tdb._backend.keys) == expected_calls
 
 
-@pytest.mark.parametrize("ch", "!@#$%^&*()~`+={}]<>,;:'\" .")
+@pytest.mark.parametrize("ch", "!@#$%^&()~`+={}]<>,;:'\" ")
```

Comment:
so we can't handle "." in filenames anymore? Is that a windows thing or has something happened?

Reply:
Windows has a nice edgecase where it can't handle a dot at the end of a directory name so it is looking like we won't be able to support `.` either. It's actually lucky that this test was written this way

I'm adding `.` and `*` back into the test now

Reply:
I need to do some more tests to whether we can still use `.` in the middle of directory names. If so we can add some checks to ensure that that weird edge case isn't triggered

Reply:
Turns out that netCDF can't handle non-alphanumeric chars at the start of variable names so this is more than just a windows issue

Reply:
Weird. To be honest, I think I'm happy to drop support for "." and "*" in filenames, that is bad practice.

We can probably just leave that netCDF error (maybe just test that it gets thrown appropriately)?

Reply:
#188 also good

Reply:
Are you also happy if we keep support for `.` for a bit longer until we plan to do a major version release? '*' was removed, but I think it is backwards compatible (given that * will still match any existing filenames including the '*')

Reply:
Ye sure